### PR TITLE
[tests-only] Adjust pact interactions for oCIS server

### PR DESCRIPTION
**Pull request by saw-jan**
part of https://github.com/owncloud/owncloud-sdk/issues/644 and https://github.com/owncloud/owncloud-sdk/issues/780

adjust interactions for oCIS:
- filesTest

### CODE REVIEW

**Review by swoichha — COMMENTED** [review]
lgtm

File: tests/filesTest.js [hunk 1]
```diff
@@ -83,9 +84,7 @@ describe('Main: Currently testing files management,', function () {
                     dProp.appendElement('d:resourcetype', '', dResourcetype => {
                       dResourcetype.appendElement('d:collection', '', '')
                     })
-                      .appendElement('d:quota-used-bytes', '', MatchersV3.number(55))
-                      .appendElement('d:quota-available-bytes', '', MatchersV3.number(3))
-                      .appendElement('d:getetag', '', MatchersV3.regex('"[a-z0-9]+"', '"5f8d0ce8c62b5"'))
+                      .appendElement('d:getetag', '', MatchersV3.string('&quot;5f8d0ce8c62b5&quot;'))
```
Comment (individual-it) [inline tests/filesTest.js:87]: I would suggest to leave regex for etags, to make sure the etags are valid, meaning are quoted `"`

File: tests/filesTest.js [hunk 2]
```diff
@@ -140,7 +139,7 @@ describe('Main: Currently testing files management,', function () {
           .appendElement(
             'd:href', '',
             MatchersV3.regex(
-              `*.\\/remote\\.php\\/webdav\\/${testFolder}\\/${item.name}`,
+              `.*\\/remote\\.php\\/webdav\\/${testFolder}\\/${item.name}(\\/)?`,
```
Comment (individual-it) [inline tests/filesTest.js:142]: good catch `*.` does not make sense

File: tests/filesTest.js [hunk 3]
```diff
@@ -151,24 +150,21 @@ describe('Main: Currently testing files management,', function () {
...
-                  MatchersV3.regex('"[a-z0-9]+"', '"3986cd55c130a4d50ff0904bf64aa27d"')
+                  MatchersV3.string('&quot;3986cd55c130a4d50ff0904bf64aa27d&quot;')
                 )
-                .appendElement('d:getcontenttype', '', 'text/plain')
+              if (item.type === 'file') {
+                dProp
+                  .appendElement('d:getcontentlength', '', MatchersV3.number(11))
+                  .appendElement('d:getcontenttype', '',
+                    MatchersV3.regex('text\\/plain(; charset=utf-8)?', 'text/plain')
+                  )
```
Comment (individual-it) [inline tests/filesTest.js:163]: no contenttype and length for folders? is that the same in oc10? if not we need to open a bug report

File: tests/filesTest.js [hunk 4]
```diff
@@ -983,19 +995,41 @@ describe('Main: Currently testing files management,', function () {
...
+      await givenUserExists(provider, config.testUser, config.testUserPassword)
+      const reqPath = path
+      let examplePath, regexPath
+      if (path !== '/') {
+        examplePath = '/' + path + '/'
+        regexPath = examplePath.replace(/\//g, '\\\\/')
+        await givenFolderExists(provider, config.testUser, config.testUserPassword, reqPath)
+      } else {
+        examplePath = path
+        regexPath = examplePath.replace(/\//g, '\\\\/')
```
Comment (individual-it) [inline tests/filesTest.js:1013]: can be moved outside of the if condition

File: tests/helpers/providerStateHelper.js [hunk 5]
```diff
@@ -286,6 +286,19 @@ const givenTagIsAssignedToFile = (provider, username, password, fileName, tagNam
...
 
+/**
+ * uploads file using TUS protocol
+ *
+ * @param {object} provider
+ * @param {string} username
+ * @param {sring} password
+ * @param {sring} fileName
+ */
+const givenFileUploadedWithTUS = (provider, username, password, fileName) => {
```
Comment (individual-it) [inline tests/helpers/providerStateHelper.js:297]: why do we need that a given step? if its a given action then we don't care how the file is uploaded, if we want to test the TUS upload then it should not be a `given` step

File: tests/providerTest.js [hunk 6]
```diff
@@ -403,6 +403,29 @@ describe('provider testing', () => {
           chai.assert.fail('Failed to assign last created tag to last created file')
         }
       }
+    },
+    'a file is uploaded using TUS protocol': (setup, parameters) => {
```
Comment (individual-it) [inline tests/providerTest.js:407]: you only the creation of a TUS file, not an actual upload 
CC @swoichha

**Review by individual-it — CHANGES_REQUESTED** [review]
it might have been good to split this PR into multiple

File: tests/filesTest.js (hunk 3 above)
Comment (saw-jan) [reply tests/filesTest.js:163]: Yeah, same for both oC10 and oCIS

File: tests/providerTest.js (hunk 6 above)
Comment (saw-jan) [reply tests/providerTest.js:407]: @individual-it @swoichha 
The tests are like `returns TUS support information when TUS headers are set for a list call`
Please, guide me what needs to be done as given for such tests.

**Comment by kiranparajuli589** [comment]
lgtm 👍🏼

File: tests/helpers/providerStateHelper.js (hunk 5 above)
Comment (saw-jan) [reply tests/helpers/providerStateHelper.js:297]: @individual-it the given test is not for TUS upload, it is to list directory and check TUS support information.

File: tests/filesTest.js [hunk 7]
```diff
@@ -151,24 +154,24 @@ describe('Main: Currently testing files management,', function () {
...
                   MatchersV3.date('EEE, d MMM yyyy HH:mm:ss z', 'Mon, 19 Oct 2020 03:50:00 GMT')
                 )
-                .appendElement('d:getcontentlength', '', MatchersV3.number(11))
                 .appendElement('d:resourcetype', '', resourceType)
                 .appendElement(
                   'd:getetag', '',
-                  MatchersV3.regex('"[a-z0-9]+"', '"3986cd55c130a4d50ff0904bf64aa27d"')
+                  MatchersV3.regex(
+                    '"[a-z0-9]+"',
+                    '&quot;3986cd55c130a4d50ff0904bf64aa27d&quot;'
```
Comment (individual-it) [inline tests/filesTest.js:162]: why do we need to use `&quot;` here and not simply `"`?

File: tests/filesTest.js [hunk 8]
```diff
@@ -627,6 +640,7 @@ describe('Main: Currently testing files management,', function () {
       })
     })
 
+    // [oCIS] 500 server error
```
Comment (individual-it) [inline tests/filesTest.js:643]: is there an issue open for that?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/filesTest.js (hunk 7 above)
Comment (saw-jan) [reply tests/filesTest.js:162]: was just a copy paste. I will update it

File: tests/filesTest.js (hunk 8 above)
Comment (saw-jan) [reply tests/filesTest.js:643]: that should be by mistake.

**Review by individual-it — APPROVED** [review]

**Comment by individual-it** [comment]
I've squashed the commits, this PR is ready to be merged when CI is green

**Pull request merged**